tests/course/test_review_view.py

```diff
@@ -6,47 +6,63 @@
+        assert Review.objects.count() == 1
+        assert b"Update Review" in response.content
+        assert b"Delete Review" in response.content
+
+    def test_display_review_not_existed(self, authorized_second_client, new_review):
+        response = authorized_second_client.get('/course/' + str(new_review.course_id), {'review_view': False})
+        assert response.status_code == 200
+        assert Review.objects.count() == 0
+        assert b"Add Review" in response.content
+
+    @pytest.mark.parametrize("rating, content", [(4, "Updated review content")])
+    def test_update_review(self, authorized_second_client, persist_review, rating, content):
+        response = authorized_second_client.post(f'/course/{persist_review.course_id}/update_review',
+                                                 {'rating': rating, 'content': content})
         assert response.status_code == 302
-        assert Review.objects.get(pk=persist_review.pk).rating == 1
-        assert Review.objects.get(pk=persist_review.pk).content == "Greate course"
+        updated_review = Review.objects.get_reviews_by_course(course=persist_review.course_id)[0]
+        assert persist_review.pk == updated_review.pk
+        assert updated_review.rating == rating
+        assert updated_review.content == content
 
-    def test_delete_review(self, client, persist_user, persist_review):
-        client.force_login(persist_user)
-        response = client.post(reverse('delete_review', args=[persist_review.pk]))
+    def test_delete_review(self, authorized_second_client, persist_review):
+        response = authorized_second_client.post(reverse('delete_review', args=[persist_review.course_id]))
         assert response.status_code == 302
         assert not Review.objects.filter(pk=persist_review.pk).exists()
 
-    def test_show_reviews_view(self, client, persist_user, persist_review):
-        client.force_login(persist_user)
-        response = client.get(reverse('reviews', args=[persist_review.course_id]))
+    def test_show_reviews_view(self, authorized_second_client, persist_review):
+        response = authorized_second_client.get('/course/' + str(persist_review.course_id))
         assert response.status_code == 200
-        assert any(persist_review.pk == review.pk for review in response.context['reviews_by_course'])
+        assert any(persist_review.pk == review.pk for review in response.context['reviews'])
 
-    def test_show_without_reviews_view(self, client, persist_user, persist_course):
-        client.force_login(persist_user)
-        response = client.get(reverse('reviews', args=[persist_course.course_id]))
+    def test_show_without_reviews_view(self, authorized_second_client, persist_course):
+        response = authorized_second_client.get('/course/' + str(persist_course.course_id))
         assert response.status_code == 200
-        assert response.context['reviews_by_course'] == []
+        assert list(response.context['reviews']) == []
```

Comment:
for readability, assign it to a named variable before making the assertion

Reply:
I will change it

Reply:
Done